### PR TITLE
👌 requirements file

### PR DESCRIPTION
**Pull request by blackhat-coder**

### CODE REVIEW

**Review by miraclx — APPROVED** [review]

**Comment by miraclx** [comment]
Hi. You should probably use different branches each based on the upstream `main` branch to implement each of these functionalities and then open PRs specific to each feature.

I mean, this PR handles the addition of the requirements file, another PR should implement shuffling.

So, you probably (definitely) should not be using your `main` branch.

**Comment by blackhat-coder** [comment]
My bad 😅
Should I remove the shuffle and then put it on another PR ? or I can start from next PR's ?

**Comment by miraclx** [comment]
This PR is opened via the `main` branch on your repo, this usually isn't advisable.

Best way forward is to;

- close this PR
- under a new branch checked out from this repo's `main` branch
  - add the requirements patch
  - open a PR to this repo's `main` from that branch
- under a new branch checked out from this repo's `main` branch
  - *start* implementing the shuffle feature
  - then you can open a *draft* PR until you complete it and submit it as an actual PR

in summary: unless PRs depend on themselves, they should branch out from the repo's main, so they're easier to merge.

Imagine, for instance, he wants to take the requirements but not the shuffle, this PR doesn't give the ability for that.

If they were independent branches based on the repo's immediate history, it would be possible.

**Comment by blackhat-coder** [comment]
Alright, would do that ASAP.
